modules/store/src/action_creator.ts

```diff
@@ -121,10 +120,8 @@ export function createAction<T extends string, C extends Creator>(
   }
 }
 
-export function props<P extends object>(): PropsReturnType<P> {
-  // the return type does not match TypePropertyIsNotAllowed, so double casting
-  // is used.
-  return ({ _as: 'props', _p: undefined! } as unknown) as PropsReturnType<P>;
+export function props<P extends object>(): { _as: 'props'; _p: P } {
```

Comment:
What about creating a `PropsConfig` type for `{ _as: 'props'; _p: P }`?

Reply:
Yes, we can do that.